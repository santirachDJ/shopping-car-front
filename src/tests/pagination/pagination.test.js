import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";

import { findElement } from "../utils/utilsEnzyme";
import Pagination from "../../components/pagination/pagination.component";
import Pager from "emerald-ui/lib/Pager";

const waitForComponentToPaint = async (wrapper) => {
  await act(async () => {
    await new Promise((resolve) => setTimeout(resolve, 0));
    wrapper.update();
  });
};
const handlerEmmiterModal = jest.fn();
describe("alert test", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Pagination 
        
        eventEmmiter={handlerEmmiterModal} total={10} offset={0} limit={8}
        />);
    waitForComponentToPaint(wrapper);
  });

  afterEach(() => {
    wrapper = {};
  });

  it("Matches the snapshot", () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('Find element componenet', () => {
    expect(findElement(wrapper, Pager).exists()).toBe(true)
})

it('Modal return data enbale',()=>{
     
    wrapper.find(Pager).at(0).props().onPageChange(10,8)
    wrapper.update()
   
    expect(handlerEmmiterModal).toBeCalled(); 
   
})
});
