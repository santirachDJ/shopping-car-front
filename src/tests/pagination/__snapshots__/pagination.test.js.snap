// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert test Matches the snapshot 1`] = `
Array [
  <Pagination
    eventEmmiter={[MockFunction]}
    limit={8}
    offset={0}
    total={10}
  />,
]
`;
